### ISSUE TITLE
Command Not Found via Xcode

### ISSUE THREAD

**Issue by Sherlouk** [stackotter/swift-bundler#21]
Hi there 👋 

Running into a little bit of an issue with using Xcode where it can't seem to find the Swift Bundler executable.

I installed the tool using mint (`mint install stackotter/swift-bundler`), created a new project following the commands on the README, and generated the Xcode support files.

If I run `swift bundler run` in Terminal it works as expected, and just for sanity, so does `swift-bundler run`.

However, if I build my project in Xcode, it compiles correctly ("Build Succeeded") but then throws an error.

> Could not launch “HelloWorld”
> The executable is missing. Please check your project settings and ensure that they produce a valid product.

If I check the logs, then it's clear the it's failing to run Swift Bundler:

```
/var/folders/54/7cl_0jq16sl8bg5xvn20skwm0000gn/T/SchemeScriptAction-SR1x7H.sh: line 2: swift-bundler: command not found
```

I've tried the standard lark of restarting Xcode and all that jazz to no avail.

Using Xcode 13.3.1 (latest non-beta), and bundler version 2.0.2 (latest).

Looking forward to giving this tool a try out. Thanks 🙂

**Comment by Sherlouk** [comment]
It does look like Mint saves the executable to /usr/local/bin but the PATH environment variable for Xcode only seems to include:

>  export PATH="/Applications/Xcode-13.3.1.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin"

Is there an extra step typically required to move the binary to a good location? Has this changed between Xcode versions?

**Comment by Sherlouk** [comment]
Confirming the above, it does look like the PATH env var is what's ultimately causing problems here.

I've updated the Post Action script to include this at the start:

> PATH=${PATH}:/usr/local/bin

This does seem to allow Xcode to find the executable and work as expected.

So I suppose the question here is now:

1. Should this workaround be added to the project?
2. Is the location where Mint installs the executable configurable?
3. Do we need to update any documentation?

I am of course happy to raise a PR if somebody can share some guidance on what the expected route is for a user 🙂  Thanks!

**Comment by Sherlouk** [comment]
Have checked and my Mint version is v0.16.0 which is not the latest version, according to the changelog they have tweaked where packages are installed which may be the root cause of this problem.

I'm going to update now and try again to see if it does indeed fix the issue.

**Comment by Sherlouk** [comment]
Confirmed: updating to v0.17.0 of Mint does not fix this issue without doing the workaround above (though this time it needs to be `$HOME/.mint/bin`)

**Comment by stackotter** [comment]
Thanks for notifying me of this issue! I fixed this exact issue a while ago but the fix meant that the binary had to be installed at a certain location, so I removed the fix and the issue didn't happen anymore. So I assumed that the latest version of Xcode had fixed this issue. It seems like I was incorrect.

When I look at my `PATH` variable in Xcode it seems to be the same as in Terminal, I wonder why yours isn't.

Which rc/env file adds mint to your `PATH`? You can check with `grep -r "\.mint" .*{env,rc}`. For me, it's added in `.zshenv`

**Comment by Sherlouk** [comment]
`export PATH=$PATH:$HOME/.mint/bin` is within my `.zshrc` file 🙂

**Comment by stackotter** [comment]
Interesting that Xcode seems to load zshenv but not zshrc. I think I have a good solution that shouldn't rely on the location of the binary. I'll add lines to source all of the zsh files that aren't sourced when zsh is running a script. That way, if it runs in terminal, it should run in xcode

**Comment by Sherlouk** [comment]
Happy to Guinea Pig any proposed changes! I won't try and more fixes to make sure it works as expected.

Thanks for the help, I understand this is a bit of a weird one and is clearly not impacting everyone.

**Comment by stackotter** [comment]
This should be fixed in v2.0.3
